asmcomp/amd64/emit.mlp

```diff
@@ -66,35 +66,64 @@ let emit_debug_info dbg =
 
 let fp = Config.with_frame_pointers
 
-(* Tradeoff between code size and code speed *)
-
-let fastcode_flag = ref true
-
-(* Layout of the stack frame *)
-let stack_offset = ref 0
+type gc_call =
+  { gc_size: int;                       (* Allocation size, in bytes *)
+    gc_lbl: label;                      (* Entry label *)
+    gc_return_lbl: label;               (* Where to branch after GC *)
+    gc_frame: label;                    (* Label of frame descriptor *)
+    gc_spacetime : (X86_ast.arg * int) option;
+    (* Spacetime node hole pointer and index *)
+  }
 
-let num_stack_slots = Array.make Proc.num_register_classes 0
+type bound_error_call =
+  { bd_lbl: label;                      (* Entry label *)
+    bd_frame: label;                    (* Label of frame descriptor *)
+    bd_spacetime : (X86_ast.arg * int) option;
+    (* As for [gc_call]. *)
+  }
 
-let prologue_required = ref false
+(* Environment *)
+type env = {
+  f : fundecl;
+  stack_offset : int;
+  call_gc_sites : gc_call list;
+  bound_error_sites : bound_error_call list;
+  bound_error_call : int;
+  symbols_defined : String.Set.t;
+  symbols_used : String.Set.t;
+}
+
+let mk_env f =
+  {
+    f;
+    stack_offset = 0;
+    call_gc_sites = [];
+    bound_error_sites = [];
+    bound_error_call = 0;
+    symbols_defined = String.Set.empty;
+    symbols_used = String.Set.empty;
```

Comment:
The `symbols_defined` and `symbols_used` fields seem to not be used yet, is that an oversight?

Reply:
They're compilation unit-wide variables, so they should not be part of the function-specific environment. They're probably left-over from earlier attempts.

Reply:
ouch, yes, I forgot to remove them, sorry, and thanks for catching it!